lib/audit.js

```diff
@@ -74,6 +80,51 @@ class Audit extends ArboristWorkspaceCmd {
       this.npm.output(result.report)
     }
   }
+
+  async auditLicenses (args) {
+    const reporter = this.npm.config.get('json') ? 'json' : 'detail'
+    const config = { // this is extremely limited and should consume user input rather than our own config
+      "licenses": {
+        "spdx": [
+          "MIT" 
+        ]    
+      }
+    }
+    await new Promise((res, rej) => {
+      const licensesPath = this.npm.prefix
+      licensee(config, licensesPath, (error, dependencies) => {
+        if (error) rej(error)
+        if (reporter === "json") {
+          this.jsonStringifyOutput(dependencies)
+        }
+        res()
+      })
+    })
+    
+  }
+
+  jsonStringifyOutput (dependencies) {
+    const loggableProperties = [
+      'name',
+      'version',
+      'approved',
+      'license',
+      'corrected',
+      'repository',
+      'homepage',
+      'author',
+      'contributors'
+    ]
+
+    const out = {}
+    for(const dep of dependencies) {
+      out[dep.name] = {}
+      for (const prop of loggableProperties) {
+        out[dep.name][prop] = dep[prop]
+      }
+    }
```

Comment:
nit:
```suggestion
    const out = Object.fromEntries(dependencies.map((dep) => [
      dep.name,
      Object.fromEntries(loggableProperties.map((prop) => [prop, dep[prop]]),
    ]))
```
altho sadly Object.fromEntries isn't in node 10, so we'd have to add `object.fromentries` to do this :-(

i still enjoyed typing it out to see the clarity improvement

Reply:
potentially an enhancement for npm@8 if node@10 support is dropped in it :)